### PR TITLE
Added hex radius in meters functionality with example.

### PR DESCRIPTION
**Pull request by chriswilley**
This PR includes functionality described in issue #14.

### CODE REVIEW

**Comment by reblace** [comment]
I took at look at the PR and have a few thoughts:

The hexbin radius (if set to an actual dimension (miles/KM)), may vary based on position. How noticeable this is will vary significantly based on the data being displayed, size of the hexbins, and the map projection being used (and Leaflet does support multiple projections). I don't know that there's a good general-purpose solution, which is one reason I'm hesitant to add this functionality directly to this plugin. With that said, I'd like to figure out a way to make the plugin flexible enough that you can add this capability relatively easily through configuration for the specific scenario where it's desired.

The function you're using to determine pixel size based on latitude will only work with the default projection. It also looks like it's using the center of the map as the latitude, so if someone starts centered on the equator and pans up to the poles, the hexbin radiuses will be pretty far off.

I don't think these things are necessarily issues, if you know that your data set has a relatively small bounding box, but again that makes the functionality really sensitive to the specific use case and data set.

So, with all that said. I'm gonna play around a bit and see what I can do to make it easier for you to do what you want by configuring the plugin rather than modifying it. Does that make sense?

**Comment by chriswilley** [comment]
I hear you. I originally placed the radius calculation routine in redraw() but didn't like how the layer behaved when panning (it caused the hexagons to blink out momentarily). That said, it might be preferable to having the radii be improperly sized as you mentioned.

In our use case, it's unlikely that anyone would pan the map significantly enough to cause this issue; having the hexagon radii fixed in meters suggests a small scale map layer and a high zoom level, but I get that you want this plugin to handle multiple use cases within reason.

The map projection issue is a thornier one. I know there are some Java tools that help with converting distances across projections (such as GeoTools). Someone [created a JavaScript project](https://github.com/jillesvangurp/geotools-js/blob/master/src/geotools.js) that does some of the same things (and was confusingly also named geotools), if that's helpful. Not sure how you deal with that (and some of the folks I'm working with agree that converting across projections is generally tricky, not just for this purpose).

Long story short, I agree with what you're saying. If I can be of help to get this closer to what you would want let me know.

Thanks.

**Comment by chriswilley** [comment]
I wanted to let you know that my thinking on this has evolved as a result of working with another guy on the mapping project I'm involved in. First of all, we found out that the equation we were using (in this PR) has a significant error (map.getCenter().lat * 180/Math.PI should be map.getCenter().lat * Math.PI/180); more importantly, since this is still an approximation even though we're using the earth's radius to three decimal places the hexagons have issues with their placement especially at high zoom levels.

So, we found ourselves reasoning this way: if we want meters per pixel, why not:

- take the lat/lon value for the map's center point (map.getCenter())
- project it into pixel space (map.latLngToContainerPoint())
- adjust the y value by a certain number of pixels (we chose zoom level + 8)
- project back into lat/lon (map.containerPointToLatLng())
- measure the distance from the first lat/lon value (val1.distanceTo(val2))

The best part about this approach is that it can be done using all Leaflet built-in functions as indicated above. Also in theory, although we haven't tested it, it should work at any projection.

I'm sure this isn't a priority for you, but I can send you the code we came up with if you'd like to see it. In any case, I would reject this pull request since it has at least one critical flaw.

Thanks again for developing this plugin.
